src/TideGrantOperation.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Drupal\tide_grant;
+
+use Drupal\Component\Utility\Random;
+use Drupal\user\Entity\Role;
+use Drupal\user\Entity\User;
+use Drupal\workflows\Entity\Workflow;
+
+/**
+ * Helper class for install/update ops.
+ */
+class TideGrantOperation {
+
+  /**
+   * Enable Editorial workflow if workflow module is enabled.
+   */
+  public function enableEditorialWorkflow() {
+    $moduleHandler = \Drupal::service('module_handler');
+    if (!\Drupal::service('config.installer')->isSyncing() && $moduleHandler->moduleExists('workflows')) {
+      $editorial_workflow = Workflow::load('editorial');
+      if ($editorial_workflow) {
+        $editorial_workflow->getTypePlugin()
+          ->addEntityTypeAndBundle('node', 'grant');
+        $editorial_workflow->save();
+      }
+    }
+  }
+
+  /**
+   * Add Grant Author user.
+   */
+  public function addGrantAuthorUser() {
+    if (empty(Role::load('grant_author'))) {
+      return;
+    }
+
+    // Create new Grant Author user.
+    $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
+    $user = User::create();
+
+    $random = new Random();
+    $pass = $random->string();
+    $mail = $random->string();
```

Comment:
Are you using the same random value for $pass and $mail?

Reply:
nope, it generates a new random one every time you call it - `$random = new Random()`